services/autoscaling/src/simcore_service_autoscaling/core/errors.py

```diff
@@ -10,6 +10,11 @@ class ConfigurationError(AutoscalingRuntimeError):
     msg_template: str = "Application misconfiguration: {msg}"
 
 
+class Ec2NotConnectedError(AutoscalingRuntimeError):
```

Comment:
THOUGHT:

```python
class AutoscalingRuntimeError(...):
   @classmethod
   @property
   def code(cls):
        return f"autoscaling.{camel_to_snake(cls.__name__)}" #<--- code created automatically

class Ec2NotConnectedError(AutoscalingRuntimeError):
         msg_template: str = "Cannot connect with ec2 server"

class Ec2InstanceNotFoundError(AutoscalingRuntimeError):
    msg_template: str = "Needed instance was not found"


```

Reply:
Perhaps the base class can be added somewhere in ``models_library.errors`` as a base class to build pydnatic-like exceptions?